accord-core/src/main/java/accord/local/Node.java

```diff
@@ -363,7 +363,7 @@ public TxnId nextTxnId(Txn.Kind rw, Domain domain)
 
     public Future<Result> coordinate(Txn txn)
     {
-        return coordinate(nextTxnId(txn.kind(), Key), txn);
+        return coordinate(nextTxnId(txn.kind(), txn.keys().domain()), txn);
```

Comment:
We don't necessarily know the `homeKey`

Reply:
Sure, but the one case where it is used is
```
        if (homeKey() != null)
            return switchOnDomain(txnId().domain(), homeKey(), PartialKeyRoute::empty, PartialRangeRoute::empty);
```
Since it is in flags sure it's basically free to have though.

Reply:
Hmm. I guess we keep `RangeDeps` and `KeyDeps` separate so this is actually implicit. So perhaps we don't need it.

Reply:
Ah, `homeKey` is presently always a `RoutingKey` so this information isn't retained in the `homeKey`.

Reply:
Hmm, that said, we could probably record it whenever we save the `homeKey` for some other reason. Or, we could stop accepting a `homeKey` on its own and require that we receive a `Route`.

Reply:
Didn't 100% follow. Home key is supposed to be some `Routable` that survives topology changes so why would we care about the domain of the transaction for the home key?

When a coordinator mints a transaction there are couple of immutable things about the transaction that are known either provided by the application or generated at the coordinator. Home key is one, txnid and its components, the txn description from the application (keys etc.).

Putting it in the txnid makes sense when someone will only have the txnid, but really needs to know the domain of the transaction?

Reply:
The home key isn't a `Routable`, it's always a `RoutingKey`

What I meant, though, was that we could stop maintaining a separate `homeKey` register in `Command`, and require that a `Route` is provided by any method that wants to provide this sort of information. This `Route` would have enough information to specify the `Domain`

Reply:
I will leave a TODO to consider this later as we address serialization/caching improvements, as I expect we will want to reform how we persist additional information like the homeKey anyway.